backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Migros_SSCC_Label.sql

```diff
@@ -0,0 +1,97 @@
+/*
+ * #%L
+ * de.metas.fresh.base
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+DROP FUNCTION IF EXISTS report.Migros_SSCC_Label(IN p_M_HU_ID numeric)
+;
+
+CREATE FUNCTION report.Migros_SSCC_Label(IN p_M_HU_ID numeric)
+    RETURNS TABLE
+            (
+                org_address     Character Varying,
+                SSCC            Character Varying,
+                p_customervalue Character Varying,
+                priceactual     numeric,
+                p_name          Character Varying,
+                cu_per_tu       numeric,
+                tu_per_lu       numeric,
+                net_weight      numeric,
+                gross_weight    numeric,
+                order_docno     Character Varying,
+                p_value         Character Varying,
+                lotcode         Character Varying,
+                paletno         Character Varying,
+                customer        Character Varying,
+                ad_language     Character Varying,
+                lotnumberdate   text
+            )
+AS
+$$
+SELECT DISTINCT *
+FROM (WITH value_check AS (SELECT COUNT(DISTINCT p_value) AS distinct_count
```

Comment:
the whole
```sql
 SELECT ....
                         FROM report.fresh_HU_SSCC_Label_Report
                         WHERE M_HU_ID = p_M_HU_ID
```

can go to a CTE. you don't have to call the fresh_HU_SSCC_Label_Report twice (here and at bottom).

Also pls keep in mind that you are running into a function, so you don't have to get everything in one query.

Reply:
also, what are you really counting with `COUNT(DISTINCT p_value)` ? what's that p_value?